upcase_booklist_app/data/users.py

```diff
@@ -0,0 +1,64 @@
+
+
+users = [
+    {
+        'user_id': 1,
+        'user_name': 'Tim McGeorge',
+        'user_bio': "I'm a software engineer looking to understand more about computer science.",
+        'user_details': [
+            {
+                'favorite_category': 'Computer Science',
+                'user_email': 'tim_mcgeorge@example.com',
```

Comment:
So I'd say that `email` belongs in the top level of each user dict, just like `name`. Then you may want to have two separate nested dicts for `category` and `books`. The latter will really be relationships once you build a database, so they should be separate here too.